src/web/app/user-join-page.component.spec.ts

```diff
@@ -106,6 +115,44 @@ describe('UserJoinPageComponent', () => {
     expect(fixture).toMatchSnapshot();
   });
 
+  it('should show error message if 4xx is returned when joining course', () => {
+    const errorMessage = '404 ERROR';
+    jest.spyOn(courseService, 'joinCourse').mockReturnValue(throwError({
+      error: {
+        message: errorMessage,
+      },
+      status: 404,
+    }));
+
+    const modalSpy = jest
+        .spyOn(simpleModalService, 'openInformationModal')
+        .mockReturnValue(createMockNgbModalRef());
+
+    component.joinCourse();
+
+    expect(modalSpy).toHaveBeenCalledTimes(1);
+    expect(modalSpy).toHaveBeenCalledWith('ERROR',
+        SimpleModalType.DANGER, errorMessage);
+  });
+
+  it('should show error message if 5xx is returned when joining course', () => {
+    const errorMessage = '502 ERROR';
+    jest.spyOn(courseService, 'joinCourse').mockReturnValue(throwError({
+      error: {
+        message: errorMessage,
+      },
+      status: 502,
+    }));
+
+    const modalSpy = jest
+        .spyOn(ngbModal, 'open')
+        .mockReturnValue(createMockNgbModalRef());
+
+    component.joinCourse();
+
+    expect(modalSpy).toHaveBeenCalledTimes(1);
```

Comment:
Should also check that the modal is initiated with the correct values (`requestId` and `errorMessage`)